coap-gateway/coap-gw.yaml

```diff
@@ -0,0 +1,68 @@
+log:
+  debug: true
+apis:
+  coap:
+    address: "0.0.0.0:5684"
+    external-port: 5684
+    fqdn: "localhost"
+    request-timeout: 10s
+    heartbeat: 4s
+    max-message-size: 1024  # default:262144
+    keepalive-enabled: true
+    keepalive-timeout: 20s
+    blockwise-transfer-enabled: true
+    blockwise-transfer-block-size: 1024
+    # tcp-signal-message-csm-disabled : false
+    # peer-tcp-signal-message-csm-disabled : false
+    # error-response-enabled : true
+    tls:
+      enabled: true
+      ca-file: "/data/certs/root_ca.crt"
+      dir-path: "/data/certs"
+      cert-file: "coap.crt"
+      key-file: "coap.key"
+      client-certificate-required: true
+clients:
+  oauth-provider:
+    oauth:
+      client-id: ""
+      client-secret: ""
+      scopes: []
+      audience: ""
+      endpoint:
+        token-url : ""
+      request-timeout : 10s
```

Comment:
move to `coap.authorization` like:
```
apis.coap.authorization:
      authority: ""
      audience: ""
```

Reply:
Give me more detail. Your are using below codes for coap.authorization but in this case OAuth config is needed but two authority and audience. 
```
// ToClientCrendtials converts to clientcredentials.Config
func (c Config) ToClientCrendtials() clientcredentials.Config {
	v := make(url.Values)
	if c.Audience != "" {
		v.Set("audience", c.Audience)
	}

	return clientcredentials.Config{
		ClientID:       c.ClientID,
		ClientSecret:   c.ClientSecret,
		Scopes:         c.Scopes,
		TokenURL:       c.Endpoint.TokenURL,
		EndpointParams: v,
	}
}
```

Reply:
You are right it is for verify token. So leave it.